components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3265,6 +3265,10 @@ void RewardsServiceImpl::OnCompleteReset(
   }
 
   std::move(callback).Run(true);
+
+  // now that we restarted everything we need to start process again
+  // as we are still on rewards page
+  StartProcess(base::DoNothing());
```

Comment:
Would it make sense to trigger the callback/observers here instead of calling `base::DoNothing()`?

Reply:
there is nothing really to call as we don't have observers for that

Reply:
I’m talking about the OnCompleteReset callback and observers.

Reply:
this are two different things, I wouldn't mix them together. @zenparsing wdyt?

Reply:
I can see how they are different concerns, although we are merging those concerns together a bit here by calling `StartProcess` (and assuming the user is on the rewards page). Haven't looked too closely, but another option for keeping them separate might be to trigger a restart of the rewards process from the rewards page reducer's `ON_COMPLETE_RESET`. Not sure if that works though.